runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/sdk/runners/DataflowPipelineTranslatorTest.java

```diff
@@ -855,43 +857,53 @@ public void populateDisplayData(DisplayData.Builder builder) {
-            (Collection<Map<String, Object>>) parDo2Properties.get("display_data");
-
-    @SuppressWarnings("unchecked")
-    Matcher<Iterable<? extends Map<String, Object>>> fn1expectedData =
-        Matchers.<Map<String, Object>>containsInAnyOrder(
-            ImmutableMap.<String, Object>builder()
-                .put("namespace", fn1.getClass().getName())
-                .put("key", "foo")
-                .put("type", "STRING")
-                .put("value", "bar")
-                .build(),
-            ImmutableMap.<String, Object>builder()
-                .put("namespace", fn1.getClass().getName())
-                .put("key", "foo2")
-                .put("type", "JAVA_CLASS")
-                .put("value", DataflowPipelineTranslatorTest.class.getName())
-                .put("shortValue", DataflowPipelineTranslatorTest.class.getSimpleName())
-                .put("label", "Test Class")
-                .put("linkUrl", "http://www.google.com")
-                .build());
-
-    @SuppressWarnings("unchecked")
-    Matcher<Iterable<? extends Map<String, Object>>> fn2expectedData =
-        Matchers.<Map<String, Object>>contains(
-            ImmutableMap.<String, Object>builder()
-                .put("namespace", fn2.getClass().getName())
-                .put("key", "foo3")
-                .put("type", "INTEGER")
-                .put("value", 1234L)
-                .build());
-    assertThat(fn1displayData, fn1expectedData);
-    assertThat(fn2displayData, fn2expectedData);
+    Collection<Map<String, String>> fn1displayData =
+            (Collection<Map<String, String>>) parDo1Properties.get("display_data");
+    Collection<Map<String, String>> fn2displayData =
+            (Collection<Map<String, String>>) parDo2Properties.get("display_data");
+
+    ImmutableSet<ImmutableMap<String, Object>> expectedFn1DisplayData = ImmutableSet.of(
+        ImmutableMap.<String, Object>builder()
+            .put("key", "foo")
+            .put("type", "STRING")
+            .put("value", "bar")
+            .put("namespace", fn1.getClass().getName())
+            .build(),
+        ImmutableMap.<String, Object>builder()
+            .put("key", "fn")
+            .put("type", "JAVA_CLASS")
+            .put("value", fn1.getClass().getName())
+            .put("shortValue", fn1.getClass().getSimpleName())
+            .put("namespace", parDo1.getClass().getName())
+            .build(),
+        ImmutableMap.<String, Object>builder()
+            .put("key", "foo2")
+            .put("type", "JAVA_CLASS")
+            .put("value", DataflowPipelineTranslatorTest.class.getName())
+            .put("shortValue", DataflowPipelineTranslatorTest.class.getSimpleName())
+            .put("namespace", fn1.getClass().getName())
+            .put("label", "Test Class")
+            .put("linkUrl", "http://www.google.com")
+            .build()
+    );
+
+    ImmutableSet<ImmutableMap<String, Object>> expectedFn2DisplayData = ImmutableSet.of(
```

Comment:
This is the version that existed in my branch before mergepocolypse. It includes a few more fields which are now being serialized, and is also a stronger assertion in that it is testing for set equality rather than subset.